include/deal.II/grid/grid_tools.h

```diff
@@ -983,6 +987,105 @@ namespace GridTools
   get_patch_around_cell(const typename Container::active_cell_iterator &cell);
 
 
+  /**
+   * This function takes a vector of active cells (hereafter named @p
+   * patch_cells) as input argument, and returns a vector of their
+   * parent cells with the coarsest common level of refinement. In
+   * other words, find that set of cells living at the same refinement
+   * level so that all cells in the input vector are children of the
+   * cells in the set, or are in the set itself.
+   *
+   * @tparam Container In C++, the compiler can not determine the type
+   * of <code>Container</code> from the function call. You need to
+   * specify it as an explicit template argument following the
+   * function name. This type has to satisfy the requirements of a
+   * mesh container (see @ref GlossMeshAsAContainer).
+   *
+   * @param[in] patch_cells A vector of active cells for which
+   *   this function finds the parents at the coarsest common
+   *   level. This vector of cells typically results from
+   *   calling the function GridTools::get_patch_around_cell().
+   * @return A list of cells with the coarsest common level of
+   *   refinement of the input cells.
+   *
+   * @author Arezou Ghesmati, Wolfgang Bangerth, 2015
+   */
+  template <class Container>
+  std::vector<typename Container::cell_iterator>
+  get_cells_at_coarsest_common_level(const std::vector<typename Container::active_cell_iterator> &patch_cells);
+
+  /**
+   * This function constructs a Triangulation (named @p
+   * local_triangulation) from a given vector of active cells. This
+   * vector (which we think of the cells corresponding to a "patch")
+   * contains active cells that are part of an existing global
+   * Triangulation. The goal of this function is to build a local
+   * triangulation that consists of only the cells given as input. The
```

Comment:
From reading below it is incorrect that it "consists of only the cells given as input".